### DISCUSSION TITLE
IOS app, cannot register new user or admin login. Backend issue?

### DISCUSSION THREAD

**Discussion by simplesisu** [wger-project/wger#1433] [Q&A]
Hi,

I've successfully installed the app on iMac with Xcode on IOS simulator. made three changes only.

Replaced `with` with `implements` on all classes in these two:
`flutter-master/lib/widgets/core/app_bar.dart`
`flutter-master/lib/widgets/workouts/app_bar.dart`

The app works fine on the iOS simulator but when trying to login or register new user I get these errors:
1. With admin I get:  `username: admin `and `password: adminadmin` 
I get this error: 
```
An error Occurred! 
non_field_errors 
Username or password unknown
```

2.  Register a new user: `username: django` and `password: thewayofwhat` `URL of the wger instance: https://wger.de`
I get this error of course:
```
An Error Occurred!
Detail
Invalid token
```

Most probably this is an backend issue, how do I solve this? Can I completely host the wger instance my self locally? if yes then how, am kind of a noob when it comes to backend questions....but I've set up setup postgresql15 and a created database, user, password etc.  

How do I make sure that the iOS connects with the backend so that I actually can register new user(only for myself) and actually start using the app and logging new data. What steps are needed? 

Please ELI5!

**Comment by rolandgeider** [comment]
Hi! Registering directly over the app is a bit difficult at the moment (also, it's not really documented 😅). The easiest way would be to register over the webapp and just use the username and password.

The easiest way to host the backend is to use the docker compose file, you just have to start it, bit if you want you can obviously also install it "natively"
